### DISCUSSION TITLE
Lead Conversion

### DISCUSSION THREAD

**Discussion by MattHCarrier** [apex-enterprise-patterns/fflib-apex-common#484] [General]
Hi, I was wondering how someone would go about implementing a Lead Conversion UoW. Currently going through fflib_ISObjectUnitOfWork and adding methods to support this functionality but wondering if since its specific to only leads would this dml be housed in Leads domain class?

**Comment by daveespo** [comment]
Hi @MattHCarrier  -- this may be the first time this request has been made in the past 5 years -- and since it's such a rarely used feature (and the semantics of what happens when the `convertLead` method is called differs from other DML operations (two objects are born out of thin air)), I can't see us adding native support for it to `fflib_SObjectUnitOfWork`

You could strap on some functionality by implementing an `IDoWork` and registering it with SOUOW ... but since it's totally disconnected from other DML operations (relationships, etc.), I don't think you're gaining much by stuffing it into an SOUOW commit operation.

**Reply by MattHCarrier** [reply]
Yeah that's where my head's at as well - ill just implement a one off method for this use case and hopefully that's good enough. Thanks for the quick reply and I appreciate everything you're doing here :)
